core-web/libs/sdk/react/src/lib/hooks/useDotcmsEditor.ts

```diff
@@ -4,26 +4,51 @@ import {
     DotCMSPageEditorConfig,
     destroyEditor,
     initEditor,
-    isInsideEditor,
-    updateNavigation
+    isInsideEditor as isInsideEditorFn,
+    updateNavigation,
+    CUSTOMER_ACTIONS,
+    postMessageToEditor
 } from '@dotcms/client';
+
+import { DotCMSPageContext } from '../models';
 export const useDotcmsEditor = (config?: DotCMSPageEditorConfig) => {
-    const [isInsideEditorPage, setIsInsideEditorPage] = useState(false);
+    const [isInsideEditor, setIsInsideEditor] = useState(false);
+    const [pageInfo, setPageInfo] = useState<DotCMSPageContext['pageAsset'] | null>(null);
+
+    const handlePostMessage = (event: MessageEvent) => {
+        const insideEditor = isInsideEditorFn();
+        if (!insideEditor) {
+            return;
+        }
+
+        if (event.data.name === 'SET_PAGE_INFO') {
+            setPageInfo(event.data.payload);
+        }
+    };
+
     useEffect(() => {
-        const insideEditor = isInsideEditor();
+        const insideEditor = isInsideEditorFn();
         if (insideEditor) {
-            initEditor(config);
+            initEditor();
             updateNavigation(config?.pathname || '/');
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.GET_PAGE_INFO,
+                payload: {
+                    pathname: config?.pathname
+                }
+            });
```

Comment:
This is acting like a ping pong, why can't we just use the ping pong we already have?

Reply:
Or... should we remove the ping pong?

Reply:
Removed ping pong and added this inside `initEditor()`